index.bs

```diff
@@ -5997,6 +6001,41 @@ leakage due to such an attack:
     - When verifying an {{AuthenticatorAssertionResponse}} response from the [=authenticator=], make it indistinguishable whether
           verification failed because the signature is invalid or because no such user or credential is registered.
 
+    - Perform a different authentication step, such as username and password authentication,
+        before initiating the WebAuthn [=authentication ceremony=].
+        This moves the username enumation problem from the WebAuthn [=authentication ceremony=]
+        to the preceding authentication step, where it may be easier to solve.
+
+
+## Privacy leak via credential IDs ## {#sctn-credential-id-privacy-leak}
+
+[INFORMATIVE]
+
+This privacy consideration applies to [=[RPS]=] supporting [=single-factor=] authentication with [=non-resident credentials=].
+
+In this case the {{PublicKeyCredentialRequestOptions/allowCredentials}} argument risks leaking [PII],
+if the user can initiate an [=authentication ceremony=] by only providing a username.
+[=Credential IDs=] are designed to not be correlatable between [=[RPS]=],
+but the length of a [=credential ID=] might be a hint as to what type of [=authenticator=] created it.
+It is likely that a user will use the same username and set of [=authenticators=] for several [=[RPS]=],
+so the number of [=credential IDs=] in {{PublicKeyCredentialRequestOptions/allowCredentials}} and their lengths
+might serve as a global correlation handle to de-anonymize the user.
+
+The simplest ways to prevent this information leak
+is to not support [=single-factor=] authentication with [=non-resident credentials=],
```

Comment:
I'm thinking that @manger [has a point](https://github.com/w3c/webauthn/pull/1250/files#r299430803) in the sense that we ought to distinguish between "sign-in"&mdash;when there is no "ambient" notion of who the user is (eg: cookies&mdash;and "re-authentication", when there _is_ an "ambient" notion of who the user is.

It is the "sign-in" case where a malicious client might be able to contact the RP server and request sign-in and receive `credentialIds` in the `allowList`, yes?  In contrast, in the "re-authn" case, a malicious client will have to have captured whatever "ambient creds" (typically cookies) that the RP employs and replay them, yes?

If the above is nominally correct, then we ought to distinguish them.   This in general harkens back to the sorts of subtleties  issue #334 is arguing we ought to clarify in the spec.

Reply:
I added mentions of session cookies as a first authentication step, what do you think?

Reply:
LGTM, thx :)